src/store/middleware.ts

```diff
@@ -0,0 +1,61 @@
+import { ThunkDispatch } from 'redux-thunk';
+import { Middleware } from 'redux';
+import { RootState } from '../reducers/index';
+import {
+  setTenant,
+  SetAppStateAction,
+  setAssetId,
+  setModelAndRevisionAndNode,
+} from '../modules/app';
+
+export const locationStateMiddleware: Middleware<
+  {},
+  RootState,
+  ThunkDispatch<any, any, SetAppStateAction>
+> = store => next => action => {
+  const response = next(action);
+  // Do stuff
+  if (
+    action.type === '@@router/LOCATION_CHANGE' &&
+    action.payload.type !== 'PUSH'
+  ) {
+    const { app } = store.getState();
+    const [, tenant, type, p1, p2, p3] = action.payload.location.pathname.split(
+      '/'
+    );
+    if (type === 'asset') {
+      const rootAssetId =
+        p1 && !Number.isNaN(parseFloat(p1)) ? Number(p1) : undefined;
```

Comment:
Are you sure this isNaN function always does what we think? See https://stackoverflow.com/questions/1303646/check-whether-variable-is-number-or-string-in-javascript for an additional requirement. I've had issues with this before :P

Also see https://stackoverflow.com/questions/25176459/is-number-isnan-more-broken-than-isnan

Reply:
Should we also extract this logic into a utils function as it is used many times?

Reply:
this is only done once so no need to extract

also the parsefloat should work the quirks 
we should not use `isNaN` (the global) as it is up to browser implementation, where as Number.isNaN is standardized.
```
Number.isNaN("asdfsad")
false
Number.isNaN(parseFloat("asdfsad"))
true
```